### ISSUE TITLE
Lazy loading in Spring Demo

### ISSUE THREAD

**Issue by denis-anisimov** [vaadin/framework8-issues#156]
Depends on #10.
Depends on new Grid and new DataSource.

Should be done one they are in the master.

I have no idea what lazy loading is in this context.
Ask @Artur

**Comment by denis-anisimov** [comment]
Create a custom Vaadin data source for this example – don't implement a generic reusable SpringDataSource for this demo.

I believe this is related to the Gird/DataSource migration.

**Comment by tsuoanttila** [comment]
While doing this we also need to add correct handling for removing items. currently disabled in a pull request.

**Comment by tsuoanttila** [comment]
Split part of the data source functionality to its own ticket. Still needs proper lazy loading implementation
